### PR TITLE
Add Request name label for every requests

### PR DESCRIPTION
**Pull request by sharath2106**
This pull request is based on issue #133 

I've added a field to Enter the request name for each request. Please find the screenshot below - 

Screenshot 2019-10-04 at 2 47 24 PM

Ideally, the same request name should be listed down in the History section of the request. The changes are also made in the corresponding files.

### CODE REVIEW

File: pages/index.vue [hunk 1]
```diff
@@ -1,5 +1,6 @@
 <template>
   <div class="page">
+      <input :class="{ error: !requestName }" @keyup.enter="requestName ? validRequestName() : null" id="label" name="label" type="label" v-model="label">
```
Comment (liyasthomas) [inline pages/index.vue]: What's the use of `validRequestName()` ?
I meant, why you want to pattern match a `label` string on every keyup?

File: pages/index.vue [hunk 2]
```diff
@@ -520,6 +522,9 @@
       }
     },
     computed: {
+      requestName() {
+        return this.label.match(/^([^?]*)\??/)[1]
+      },
```
Comment (liyasthomas) [inline pages/index.vue]: Label will be a string which user decides, we don't want to enforce any pattern for it. Isn't it?

File: pages/index.vue [hunk 3]
```diff
@@ -686,6 +691,13 @@
           behavior: 'smooth'
         });
       },
+      async validRequestName() {
+        if (!this.requestName) {
+          this.$toast.error('Request Name is not valid', {
+            icon: 'error'
+          });
+          return;
+        }},
```
Comment (liyasthomas) [inline pages/index.vue]: Can you tell me an use case for this?

File: pages/index.vue [hunk 4]
```diff
@@ -773,6 +785,7 @@
 
             // Addition of an entry to the history component.
             const entry = {
+              label: this.requestName,
```
Comment (liyasthomas) [inline pages/index.vue]: Already `label` is binded with input field value isn't it? so we don't need to specify `this.name` for it.

**Comment by sharath2106** [comment]
@liyasthomas Apologies. I just committed and pushed from the wrong repo. I'll make the changes and push again.

**Pull request merged**